secp256k1-sys/build.rs

```diff
@@ -16,6 +16,41 @@ use std::env;
 fn main() {
     // Actual build
     let mut base_config = cc::Build::new();
+    
+    // On riscv32, we need to build C libraries using the riscv-gnu-toolchain and
+    // clang for compiling C code.
+    if env::var("CARGO_CFG_TARGET_ARCH").unwrap() == "riscv32" {
+        const DEFAULT_RISCV_GNU_TOOLCHAIN: &str = "/opt/riscv";
+        println!("cargo:rerun-if-env-changed=RISCV_GNU_TOOLCHAIN");
+
+        let riscv_gnu_toolchain_path = env::var("RISCV_GNU_TOOLCHAIN").unwrap_or_else(|_| {
+            println!("cargo:warning=Variable RISCV_GNU_TOOLCHAIN unset. Assuming '{DEFAULT_RISCV_GNU_TOOLCHAIN}'");
+            println!("cargo:warning=Please make sure to build riscv toolchain:");
+            println!("cargo:warning=  git clone https://github.com/riscv-collab/riscv-gnu-toolchain && cd riscv-gnu-toolchain");
+            println!("cargo:warning=  export RISCV_GNU_TOOLCHAIN={DEFAULT_RISCV_GNU_TOOLCHAIN}");
+            println!("cargo:warning=  configure --prefix=\"$RISCV_GNU_TOOLCHAIN\" --with-arch=rv32im --with-abi=ilp32");
+            println!("cargo:warning=  make -j$(nproc)");
+
+            // if unset, try the default and fail eventually
+            DEFAULT_RISCV_GNU_TOOLCHAIN.into()
+        });
+
+        base_config
+            .compiler("clang")
+            .no_default_flags(true)
+            .flag(&format!("--sysroot={riscv_gnu_toolchain_path}/riscv32-unknown-elf"))
+            .flag(&format!("--gcc-toolchain={riscv_gnu_toolchain_path}"))
+            .flag("--target=riscv32-unknown-none-elf")
+            .flag("-march=rv32im")
+            .flag("-mabi=ilp32")
+            .flag("-mcmodel=medany")
+            .flag("-Os")
```

Comment:
there were some issues when building the 1.79 rust toolchain so it'd be good to test with that